### PR TITLE
Fix persistance issue with directory

### PR DESCRIPTION
**Pull request by LeoRuiz01**
Previous PR did not update the directory after it was ran, this PR fixes that issue adding a flag that lets the user specify the directory that the oauth2l must look at.  It also modifies the README to include said changes.

### CODE REVIEW

**Comment by shinfan** [comment]
Please fix the tests.

File: README.md [hunk 1]
```diff
@@ -382,6 +381,22 @@ Path to Curl CLI. For optional use with "curl" command.
 $ oauth2l curl --curlcli /usr/bin/curl --type sso --email me@google.com --scope cloud-platform --url https://pubsub.googleapis.com/v1/projects/my-project-id/topics
 ```
 
+## web --stop
```
Comment (andyrzhao) [inline README.md:384]: Please also document "web" as a top-level command, after "reset" section.

File: go.mod [hunk 2]
```diff
@@ -6,6 +6,7 @@ require (
 	cloud.google.com/go v0.48.0
 	github.com/golang/protobuf v1.3.2
 	github.com/jessevdk/go-flags v1.4.0
+	github.com/spf13/viper v1.7.1 // indirect
```
Comment (andyrzhao) [inline go.mod:9]: remove viper here

File: main.go [hunk 3]
```diff
@@ -117,8 +117,8 @@ type resetOptions struct {
 
 // Options for "web" command
 type webOptions struct {
-	// Stop the web app
-	Stop bool `long:"stop" description:"Stops the OAuth2l Playground."`
+	Stop      bool   `long:"stop" description:"Stops the OAuth2l Playground."`
+	Directory string `long:"directory" description:"Sets the directory of where OAuth2l-web located. Defaults to ~/.oauth2l-web." `
```
Comment (andyrzhao) [inline main.go:121]: "where OAuth2l-web should be located"

File: main.go [hunk 4]
```diff
@@ -378,10 +378,17 @@ func main() {
 
 		os.Exit(task(token))
 	} else if cmd == "web" {
+		var directory string = "~/.oauth2l"
```
Comment (andyrzhao) [inline main.go:381]: Let's simplify the if statements by updating "directory" with opts.Web.Directory once at the very beginning if not empty, then use "directory" for the rest of this section.

Comment (andyrzhao) [inline main.go:381]: this should be ~/.oauth2l-web

File: main.go [hunk 5]
```diff
@@ -378,12 +378,21 @@ func main() {
 
 		os.Exit(task(token))
 	} else if cmd == "web" {
-		if opts.Web.Stop {
-			util.WebStop()
+		var directory string = "~/.oauth2l-web"
+		if len(opts.Web.Directory) > 0 {
```
Comment (andyrzhao) [inline main.go:382]: No need for the extra nested if statements. Simplify to:

if len(opts.Web.Directory) > 0 {
directory = opts.Web.Directory
}
if opts.Web.Stop {
	util.WebStop(directory)
} else {
	util.Web(directory)
}

File: README.md [hunk 6]
```diff
@@ -199,6 +198,30 @@ file `~/.oauth2l` by default.
...
 
+### web
+
+Runs the web application and opens a new window to show it. If the web application is not installed it is installed in `~/.oauth2l-web` by default.
+
+```bash
+$ oauth2l web
+```
+
+## web --stop
```
Comment (andyrzhao) [inline README.md:209]: Move the documentation for "web --stop" and "web --directory" back to the command options section where you had it before : )

File: main.go [hunk 7]
```diff
@@ -117,8 +117,8 @@ type resetOptions struct {
 
 // Options for "web" command
 type webOptions struct {
-	// Stop the web app
-	Stop bool `long:"stop" description:"Stops the OAuth2l Playground."`
+	Stop      bool   `long:"stop" description:"Stops the OAuth2l Playground where OAuth2l-web s."`
```
[the remark this answers is not in the document]
Comment (LeoRuiz01) [reply main.go:120]: Yep sorry!

File: main.go (hunk 5 above)
Comment (LeoRuiz01) [reply main.go:382]: I was completely over complicating it, thanks for the suggestion!

File: README.md (hunk 6 above)
Comment (LeoRuiz01) [reply README.md:209]: Fixed!

**Pull request merged**